rclcpp/include/rclcpp/topic_statistics/subscriber_topic_statistics.hpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__TOPIC_STATISTICS__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+#define RCLCPP__TOPIC_STATISTICS__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rcl/time.h"
+#include "rclcpp/time.hpp"
+#include "rclcpp/publisher.hpp"
+#include "rclcpp/timer.hpp"
+
+#include "metrics_statistics_msgs/msg/metrics_message.hpp"
+
+#include "libstatistics_collector/collector/generate_statistics_message.hpp"
+#include "libstatistics_collector/topic_statistics_collector/constants.hpp"
+#include "libstatistics_collector/topic_statistics_collector/received_message_period.hpp"
+#include "libstatistics_collector/moving_average_statistics/types.hpp"
+
+namespace
+{
+/// Return the current nanoseconds (count) since epoch
+/**
+ * For now, use hard coded time instead of a node's clock (to support sim time and playback)
+ * due to node clock lifecycle issues.
+ * @return the current nanoseconds (count) since epoch
+ */
+int64_t GetCurrentNanosecondsSinceEpoch()
```

Comment:
This style doesn't follow the rest of rclcpp, it should be `get_current_nanoseconds_since_epoch()`. There are lots of instances of this in the new code. I believe it should be changed to be consistent with rclcpp.

Reply:
Will do. The style you're seeing is from the [Google Style Guide](https://google.github.io/styleguide/cppguide.html#Function_Names); I'll make the necessary changes to be consistent with the `rclcpp` style.

Reply:
When contributing to a core ROS 2 package, refer to the ROS 2 style guide:

https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#function-and-method-naming